Client/src/GameContext/ClientEntityPool.cpp

```diff
@@ -2,31 +2,64 @@
 // Created by hippolyteb on 11/29/16.
 //
 
+#include <Messages/ReceivedNetworkPayloadMessage.hpp>
+#include <Json/Json.hpp>
 #include "ClientEntityPool.hpp"
 
 void ClientEntityPool::Draw(sf::RenderTexture &target, TextureBag &bag) {
     for (const auto& i : _pool)
     {
-        if (i->ImplementTrait(Trait::Drawable)) {
-            IDrawable *entity = dynamic_cast<IDrawable*>(i.GetInstance());
+        if (i.second->ImplementTrait(Trait::Drawable)) {
```

Comment:
The `_pool` is now a `std::set`, which behave (not quite) like a map. `first` is a key, `second` is the value.